diff --git a/framework/cauldron/framework/src/render/vk/pipelineobject_vk.cpp b/framework/cauldron/framework/src/render/vk/pipelineobject_vk.cpp
index 31cd297..03933f6 100644
--- a/framework/cauldron/framework/src/render/vk/pipelineobject_vk.cpp
+++ b/framework/cauldron/framework/src/render/vk/pipelineobject_vk.cpp
@@ -290,19 +290,18 @@ namespace cauldron
 
         // viewport and scissors are dynamic stage to avoid recreating the pipeline if the size of the render target changes
         // Topology is also dynamic
-        VkDynamicState dynamicStates[4];
-        uint32_t numDynamicStates = 0;
-        dynamicStates[numDynamicStates++] = VK_DYNAMIC_STATE_VIEWPORT;
-        dynamicStates[numDynamicStates++] = VK_DYNAMIC_STATE_SCISSOR;
+        std::vector<VkDynamicState> dynamicStates;
+        dynamicStates.push_back(VK_DYNAMIC_STATE_VIEWPORT);
+        dynamicStates.push_back(VK_DYNAMIC_STATE_SCISSOR);
 #ifdef VK_EXT_extended_dynamic_state
-        dynamicStates[numDynamicStates++] = VK_DYNAMIC_STATE_PRIMITIVE_TOPOLOGY_EXT;
+        dynamicStates.push_back(VK_DYNAMIC_STATE_PRIMITIVE_TOPOLOGY_EXT);
 #endif
         VkPipelineDynamicStateCreateInfo dynamicStateInfo;
         dynamicStateInfo.sType = VK_STRUCTURE_TYPE_PIPELINE_DYNAMIC_STATE_CREATE_INFO;
-        dynamicStateInfo.pNext = 0;
+        dynamicStateInfo.pNext = nullptr;
         dynamicStateInfo.flags = 0;
-        dynamicStateInfo.dynamicStateCount = numDynamicStates;
-        dynamicStateInfo.pDynamicStates = dynamicStates;
+        dynamicStateInfo.dynamicStateCount = static_cast<uint32_t>(dynamicStates.size());
+        dynamicStateInfo.pDynamicStates = dynamicStates.data();
 
         // build shaders
         PipelineShadersBuilder pipelineShaders(GetDevice(), pipelineDesc);
